.github/workflows/native-wsl.yml

```
@@ -0,0 +1,153 @@
+          npm install
+          npm install @typescript-eslint/parser@4 --no-save
+        fi
+        if [ ${{ matrix.node-version }} -ge 12 ]; then
+          npm install
+        fi
+        npm run copy-metafiles
+        npm run pretest
+        npm run tests-only
+    - name: Install dependencies for Node <= 10
+      if: matrix.node-version <= '10' && matrix.configuration == 'native'
+      run: |
+        npm install --legacy-peer-deps
+        npm install eslint@7 --no-save
+    - name: Install dependencies for Node > 10
+      if: matrix.node-version > '10'  && matrix.configuration == 'native'
+      run: npm install
+    - name: Install the latest version of nyc
+      if: matrix.configuration == 'native'
+      run: npm install nyc@latest --no-save
+    - name: Copy metafiles for Node <= 8
+      if: matrix.node-version <= 8 && matrix.configuration == 'native'
+      env:
+        ESLINT_VERSION: 6
+        TRAVIS_NODE_VERSION: ${{ matrix.node-version }}
+      run: |
+        npm run copy-metafiles
+        bash ./tests/dep-time-travel.sh 2>&1
+    - name: Copy metafiles for Node > 8
+      if: matrix.node-version > 8 && matrix.configuration == 'native'
+      env:
+        ESLINT_VERSION: 7
+        TRAVIS_NODE_VERSION: ${{ matrix.node-version }}
+      run: |
+        npm run copy-metafiles
+        bash ./tests/dep-time-travel.sh 2>&1
+    - name: Config git to allow symlinks
+      if: matrix.configuration == 'native'
+      run: |
+        git config core.symlinks true
+        git reset --hard
+    - name: Install ./resolver dependencies in Native
+      if: matrix.configuration == 'native'
+      shell: pwsh
+      run: |
+        npm config set package-lock false
+        $resolverDir = "./resolvers"
+        Get-ChildItem -Directory $resolverDir |
+          ForEach {
+            Write-output $(Resolve-Path $(Join-Path $resolverDir  $_.Name))
+            Push-Location $(Resolve-Path $(Join-Path $resolverDir  $_.Name))
+              npm install
+              npm ls nyc > $null;
+              if ($?) {
+                npm install nyc@latest --no-save
+              }
+              Pop-Location
+            }
+        npm config set package-lock true
+    - name: Run tests in Native
+      if: matrix.configuration == 'native'
+      shell: pwsh
+      run: |
+        npm run pretest
+        npm run tests-only
+        $resolverDir = "./resolvers";
+        $resolvers = @();
+        Get-ChildItem -Directory $resolverDir |
+          ForEach {
+            $resolvers += "$(Resolve-Path $(Join-Path $resolverDir $_.Name))";
+          }
+        $env:RESOLVERS = [string]::Join(";", $resolvers);
+        foreach ($resolver in $resolvers) {
+            Set-Location -Path $resolver.Trim('"')
+            npm test
+            Set-Location -Path $PSScriptRoot
+        }
+    - name: Upload Codecov
+      uses: codecov/codecov-action@v4
+      with:
+        token: ${{ secrets.CODECOV_TOKEN }} # not required for public repos
```

```suggestion
      uses: codecov/codecov-action@v3.1.5
      with:
```
i want to stick with tokenless, and this is the only tokenless version that uses node20 instead of node16